src/plugins/content_management/server/core/crud.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { EventBus } from './event_bus';
+import type { ContentStorage, StorageContext } from './types';
+
+export class ContentCrud implements ContentStorage {
+  private storage: ContentStorage;
+  private eventBus: EventBus;
+  public contentType: string;
+
+  constructor(
+    contentType: string,
+    contentStorage: ContentStorage,
+    {
+      eventBus,
+    }: {
+      eventBus: EventBus;
+    }
+  ) {
+    this.contentType = contentType;
+    this.storage = contentStorage;
+    this.eventBus = eventBus;
+  }
+
+  public async get<Options extends object = object, O = any>(
```

Comment:
Would be great to make API responses a wrapping of nested objects, which will allow for extensions without breaking changes in the future, some thing like this:

```suggestion
  public async get<Options extends object = object, O = GetMethodResult>(
  
  interface GetMethodResult {
    item: ItemResult;
    // Place for future fields.
  }
  
  interface ItemResult {
    item: ContentItem; // See "ContentItem" from above comment.
    // Place for future fields
  }
```

for example, in the future it could support stale data checks:

```ts
  type ItemResult = 
    | ItemResultFull
    | ItemResultNoChanges.

  interface ItemResultFull {
    item: ContentItem;
  }

  interface ItemResultNoChanges {
    noChanges: true;
  }
```

then these interfaces can also be reused in other methods:

```ts
  bulkGet(/* .. */): Promise<{ items: ItemResult }>;
  subscribe(/* .. */): Observable<ItemResult>;
  search(/* .. */): Promise<{ items: ItemResult }>;  
```

Reply:
> Also as part of these apis, I think it is better to fallback O to unknown instead of any unless there is a good reason not to

It does not work with `unknown`. TS complains that `whatever is passed as generic` does not satisfies `unknown`. But if you manage to make it work, happy to change 😊

Reply:
> It does not work with unknown. TS complains that whatever is passed as generic does not satisfies unknown. But if you manage to make it work, happy to change

got it, thanks for explaining

Reply:
> Would be great to make API responses a wrapping of nested objects

Good idea, I added it in https://github.com/elastic/kibana/pull/150692/commits/fd6f7b97f3e8ef12cdf9f99ebe7f9a40ed84bd10